Lib/test/test_pty.py

```diff
@@ -142,47 +136,47 @@ def test_openpty(self):
             mode = None
 
         new_stdin_winsz = None
-        if self.stdin_rows != None and self.stdin_cols != None:
+        if self.old_stdin_winsz != None:
             try:
                 # Modify pty.STDIN_FILENO window size; we need to
                 # check if pty.openpty() is able to set pty slave
                 # window size accordingly.
-                debug("Setting pty.STDIN_FILENO window size")
-                debug(f"original size: (rows={self.stdin_rows}, cols={self.stdin_cols})")
-                target_stdin_rows = self.stdin_rows + 1
-                target_stdin_cols = self.stdin_cols + 1
-                debug(f"target size: (rows={target_stdin_rows}, cols={target_stdin_cols})")
-                target_stdin_winsz = struct.pack("HHHH", target_stdin_rows,
-                                                 target_stdin_cols, 0, 0)
-                _set_term_winsz(pty.STDIN_FILENO, target_stdin_winsz)
-
-                # Were we able to set the window size
-                # of pty.STDIN_FILENO successfully?
-                new_stdin_winsz = _get_term_winsz(pty.STDIN_FILENO)
+                debug("setting pty.STDIN_FILENO window size")
+                debug(f"original winsize: {self.old_stdin_winsz}")
+                target_stdin_winsz = tty.winsize()
+                target_stdin_winsz.ws_row = self.old_stdin_winsz.ws_row + 1
+                target_stdin_winsz.ws_col = self.old_stdin_winsz.ws_col + 1
+                target_stdin_winsz.ws_xpixel = self.old_stdin_winsz.ws_xpixel + 1
+                target_stdin_winsz.ws_ypixel = self.old_stdin_winsz.ws_ypixel + 1
+                debug(f"target winsize: {target_stdin_winsz}")
+                target_stdin_winsz.tcsetwinsize(pty.STDIN_FILENO)
+
```

Comment:
These parts of the test were written by me. Now I am rewriting everything using the api that I am intending to add to tty.py; this increases code re-usability.